source/websocket.c

```diff
@@ -70,6 +74,13 @@ struct aws_websocket {
 
         struct aws_websocket_decoder decoder;
 
+        /* Amount to increment window after a channel message has been processed. */
+        size_t incoming_message_window_update;
+
+        /* Frame currently being decoded. */
+        struct aws_websocket_incoming_frame incoming_frame;
+        bool has_incoming_frame;
```

Comment:
Pointer to structure? Set pointer when there's an incoming frame, clear when not, only access through pointer?

Reply:
cute. I'll do that